### ISSUE TITLE
confusing dataset creation process

### ISSUE THREAD

**Issue by vnoel** [pydata/xarray#1249]
In another issue I create a simple dataset like so:

```python
lat = np.random.rand(50000) * 180 - 90
lon = np.random.rand(50000) * 360 - 180
d = xr.Dataset({'latitude':lat, 'longitude':lon})
```

I expected `d` to contain two variables (`latitude` and `longitude`) with no coordinates. Instead `d` appears to contain two coordinates and no variables:

```
In [5]: d
Out[5]: 
<xarray.Dataset>
Dimensions:    (latitude: 50000, longitude: 50000)
Coordinates:
  * latitude   (latitude) float64 -76.0 -84.36 26.69 66.44 -37.85 50.13 ...
  * longitude  (longitude) float64 -148.7 -74.82 18.37 117.7 80.63 12.25 ...
Data variables:
    *empty*
```

Is this desired behavior?

**Comment by MaximilianR** [comment]
I agree that outcome isn't what I would have expected. If you put the input into a `DataArray` first, this performs as expected. 

```python
In [5]: lat = np.random.rand(50000) * 180 - 90
   ...: lon = np.random.rand(50000) * 360 - 180
   ...: d = xr.Dataset({'latitude':xr.DataArray(lat), 'longitude':lon})
   ...:
   ...:

In [6]: d
Out[6]:
<xarray.Dataset>
Dimensions:    (dim_0: 50000, longitude: 50000)
Coordinates:
  * longitude  (longitude) float64 -80.73 -118.8 88.66 -132.1 -74.87 27.2 ...
Dimensions without coordinates: dim_0
Data variables:
    latitude   (dim_0) float64 47.59 26.57 30.56 -12.52 65.44 70.32 84.45 ...
```

**Comment by shoyer** [comment]
This was intentional, but is perhaps too magical/overloaded.

Here's the reasoning:
1. If dimension names are not provided for a 1D array, presume that it's sole dimension is the same as it's name.
2. Variables with the same name as a dimension must be coordinates in the xarray data model. Hence make these variables coordinates, even if they appears in the `data_vars` argument. (This is a bit of a hold from before there was a separate `coords` argument.)

Note that if you put an array with 2 or more dimensions directly into `data_vars`/`coords` you currently get `ValueError`.

Alternative behaviors that might make more sense:
- Raise an error for this behavior, only allowing implicit dimension names when you put an array in `coords`.
- Deprecate implicit dimension names for 1D arrays altogether. Later, switch to labeling arrays with automatic dimension names like the `DataArray` constructor.

I'm not a huge fan of this second option because if you don't get an error it would be easy to construct datasets with some variables having the wrong dimension names.

**Comment by vnoel** [comment]
I don't understand the "raise an error" option. Would it mean that creating a Dataset with 1D arrays with no named dimension would raise an error? That would be very impractical.

**Comment by fmaussion** [comment]
> Would it mean that creating a Dataset with 1D arrays with no named dimension would raise an error? That would be very impractical.

it's already the case with any array of more than 1 dim.

**Comment by vnoel** [comment]
ok, I see. In case there's a vote, I vote for raising an error in 1D too :-)
